src/atomate2/abinit/schemas/core.py

```diff
@@ -0,0 +1,212 @@
+"""Core definitions of Abinit calculations documents."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Type, TypeVar, Union
+
+from abipy.abio.inputs import AbinitInput
+from abipy.electrons.gsr import GsrFile
+from abipy.flowtk import events
+from abipy.flowtk.utils import Directory, File
+from jobflow.utils import ValueEnum
+from pydantic import BaseModel, Field
+from pymatgen.core.structure import Structure
+
+from atomate2.abinit.files import load_abinit_input
+from atomate2.abinit.sets.base import AbinitInputSet
+from atomate2.abinit.utils.common import (
+    LOG_FILE_NAME,
+    MPIABORTFILE,
+    OUTDIR_NAME,
+    OUTPUT_FILE_NAME,
+    PostProcessError,
+)
+from atomate2.common.schemas.structure import StructureMetadata
+
+_T = TypeVar("_T", bound="AbinitTaskDocument")
+
+logger = logging.getLogger(__name__)
+
+
+class Status(ValueEnum):
+    """Abinit calculation state."""
+
+    # TODO: merge this somewhere with vasp => common calculation schema ?
+
+    SUCCESS = "successful"
+    FAILED = "failed"
+
+
+class JobMetadata(BaseModel):
+    """Definition of job metadata fields."""
+
+    dir_name: str = Field(None, description="The directory of this job.")
+    calc_type: str = Field(None, description="The type of calculation of this job.")
+
+
+class AbinitJobSummary(JobMetadata):
+    """Definition of summary information about an Abinit Job."""
+
+    # restart_info: RestartInfo = Field(
+    #     None, description="Restart information for the next job."
+    # )
+    # history: JobHistory = Field(None, description="Job history.")
+    abinit_input_set: AbinitInputSet = Field(
+        None, description="AbinitInputSet object used to perform calculation."
+    )
+    structure: Structure = Field(
+        None, description="Final structure of the calculation."
+    )
+    energy: float = Field(None, description="Final energy of the calculation.")
+
+
+class AbinitTaskDocument(StructureMetadata):
+    """Definition of task document about an Abinit Job."""
+
+    state: Status = Field(None, description="State of this job.")
+    run_number: int = Field(None, description="Run number of this job.")
+    dir_name: str = Field(None, description="The directory of this job.")
+    event_report: events.EventReport = Field(
+        None, description="Event report of this abinit job."
+    )
+    task_label: str = Field(None, description="The label for this job/task.")
+    abinit_input: AbinitInput = Field(
+        None, description="AbinitInput used to perform calculation."
+    )
+
+    @classmethod
+    def from_directory(
+        cls: Type[_T],
+        dir_name: Union[Path, str],
+        source: str = "log",
+        critical_events=None,
+        run_number=1,
+        structure_fixed=True,
```

Comment:
Indeed, I think it makes sense to simplify. About getting the structure, you would first try to get it from the run.abo (using `abipy.abio.outputs.AbinitOutputFile.final_structure`), then GSR, then only if None of these work, from the AbinitInput ?

Reply:
I removed the structure_fixed. Now if the calculation is killed due to walltime, I think we have to distinguish the cases. If it is a relaxation or md we can try to continue. But for the reset I would say something has to be done by the user (e.g. increase wall time, or do something so that the calculation fits in the wall  time provided). What do you think ?
Currently, I try to get the structure from run.abo or from the GSR (not from the abinit_input), otherwise when it gets killed basically you can always restart, but from the exact same structure every time...